### ISSUE TITLE
Implement glyph composition functions

### ISSUE THREAD

**Issue by madig** [trufont/trufont#172]
FontForge comes with functions to build diacritics and ligatures. That would be useful to have. And it would be even more useful if glyphs with multiple marks would respect *all* anchors of the base glyph instead of expecting anchors on all composited glyphs that additional marks are attached to, like U+1E5C. All changes to base glyphs and marks should update all composite glyphs, of course.

**Comment by pathumego** [comment]
Also, enable adding new composition instances to the library using glyph names.

**Comment by madig** [comment]
Clarification: I can compose glyphs using the "add component" function, but it doesn't respect anchors.

**Comment by davelab6** [comment]
@madig would https://github.com/typemytype/GlyphConstruction satisfy your need?

**Comment by pathumego** [comment]
@davelab6 Yeah for me it seems good. A similar thing to maintaining side bearing values as formulas would be nice too :) In glyphs you can write simple formulas with glyph names.

**Comment by madig** [comment]
Hm, I guess that does the job and more.
